File: Datos/LocalidadAD.vb
Imports Entidades
Imports System.Configuration
Imports System.Data.SqlClient
Imports Excepciones

Public Class LocalidadAD
    Public Shared Sub AltaLocalidad(Localidad As LocalidadEN)
        Using Cnn As New SqlConnection(ConfigurationManager.ConnectionStrings("Mercader").ToString())
            Cnn.Open()
            Dim ConsultaLoc As String = "INSERT INTO Localidad(Provincia_CodProvincia,Descripcion,CodigoPostal) VALUES" & _
                                    "(@CodProv,@Descripcion,@CodigoPostal)"
            Dim Cmd As New SqlCommand(ConsultaLoc, Cnn)
            Cmd.Parameters.AddWithValue("@CodProv", Localidad.CodProvincia)
            Cmd.Parameters.AddWithValue("@Descripcion", Localidad.Descripcion)
            Cmd.Parameters.AddWithValue("@CodigoPostal", Localidad.CodigoPostal)

            Cmd.ExecuteNonQuery()

        End Using
    End Sub

    Public Shared Function CargarProvincia() As List(Of ProvinciaEN)
        Dim ListaProv As New List(Of ProvinciaEN)
        Using Cnn As New SqlConnection(ConfigurationManager.ConnectionStrings("Mercader").ToString())
            Cnn.Open()

            Dim ConsultaProv As String = "SELECT CodProvincia,Descripcion FROM Provincia"
                Dim Cmd As New SqlCommand(ConsultaProv, Cnn)
                Dim Lector As SqlDataReader = Cmd.ExecuteReader()

                While Lector.Read()
                    Dim UnaProv As New ProvinciaEN
                    UnaProv.CodProvincia = Lector(0)
                    UnaProv.Descripcion = Lector(1)

                    ListaProv.Add(UnaProv)
                End While

                Return ListaProv
        End Using
    End Function

    Public Shared Function ValidarLocalidad(Nombre As String) As Integer
        Using Cnn As New SqlConnection(ConfigurationManager.ConnectionStrings("Mercader").ToString())
            Cnn.Open()

            Dim ConsultaLoc As String = "SELECT COUNT(*) FROM Localidad WHERE Descripcion=@Param1"
                Dim Cmd As New SqlCommand(ConsultaLoc, Cnn)
                Cmd.Parameters.AddWithValue("@Param1", Nombre)
                Dim Resultado As Integer = Cmd.ExecuteScalar()

                Return Resultado

        End Using
    End Function

    Public Shared Function CargarLocalidad() As List(Of LocalidadEN)
        Dim ListaLoc As New List(Of LocalidadEN)
        Using Cnn As New SqlConnection(ConfigurationManager.ConnectionStrings("Mercader").ToString())
            Cnn.Open()

                Dim ConsultaLoc As String = "SELECT CodLoc,Provincia_CodProvincia,Descripcion,CodigoPostal FROM Localidad"
                Dim Cmd As New SqlCommand(ConsultaLoc, Cnn)

                Dim Lector As SqlDataReader = Cmd.ExecuteReader()

                While Lector.Read()
                    Dim UnaLocalidad As New LocalidadEN
                    UnaLocalidad.CodLoc = Lector(0)
                    UnaLocalidad.CodProvincia = Lector(1)
                    UnaLocalidad.Descripcion = Lector(2)
                    UnaLocalidad.CodigoPostal = Lector(3)

                    ListaLoc.Add(UnaLocalidad)
                End While

                Return ListaLoc
        End Using
    End Function

End Class

File: Datos/LocalidadAD.cs
using System.Collections.Generic;
using System.Configuration;
using System.Data.SqlClient;
using Entidades;
using Microsoft.VisualBasic.CompilerServices;

namespace Datos
{
    public class LocalidadAD
    {
        public static void AltaLocalidad(LocalidadEN Localidad)
        {
            using (var Cnn = new SqlConnection(ConfigurationManager.ConnectionStrings["Mercader"].ToString()))
            {
                Cnn.Open();
                string ConsultaLoc = "INSERT INTO Localidad(Provincia_CodProvincia,Descripcion,CodigoPostal) VALUES" + "(@CodProv,@Descripcion,@CodigoPostal)";
                var Cmd = new SqlCommand(ConsultaLoc, Cnn);
                Cmd.Parameters.AddWithValue("@CodProv", Localidad.CodProvincia);
                Cmd.Parameters.AddWithValue("@Descripcion", Localidad.Descripcion);
                Cmd.Parameters.AddWithValue("@CodigoPostal", Localidad.CodigoPostal);
                Cmd.ExecuteNonQuery();
            }
        }

        public static List<ProvinciaEN> CargarProvincia()
        {
            var ListaProv = new List<ProvinciaEN>();
            using (var Cnn = new SqlConnection(ConfigurationManager.ConnectionStrings["Mercader"].ToString()))
            {
                Cnn.Open();
                string ConsultaProv = "SELECT CodProvincia,Descripcion FROM Provincia";
                var Cmd = new SqlCommand(ConsultaProv, Cnn);
                var Lector = Cmd.ExecuteReader();
                while (Lector.Read())
                {
                    var UnaProv = new ProvinciaEN();
                    UnaProv.CodProvincia = Conversions.ToInteger(Lector[0]);
                    UnaProv.Descripcion = Conversions.ToString(Lector[1]);
                    ListaProv.Add(UnaProv);
                }

                return ListaProv;
            }
        }

        public static int ValidarLocalidad(string Nombre)
        {
            using (var Cnn = new SqlConnection(ConfigurationManager.ConnectionStrings["Mercader"].ToString()))
            {
                Cnn.Open();
                string ConsultaLoc = "SELECT COUNT(*) FROM Localidad WHERE Descripcion=@Param1";
                var Cmd = new SqlCommand(ConsultaLoc, Cnn);
                Cmd.Parameters.AddWithValue("@Param1", Nombre);
                int Resultado = Conversions.ToInteger(Cmd.ExecuteScalar());
                return Resultado;
            }
        }

        public static List<LocalidadEN> CargarLocalidad()
        {
            var ListaLoc = new List<LocalidadEN>();
            using (var Cnn = new SqlConnection(ConfigurationManager.ConnectionStrings["Mercader"].ToString()))
            {
                Cnn.Open();
                string ConsultaLoc = "SELECT CodLoc,Provincia_CodProvincia,Descripcion,CodigoPostal FROM Localidad";
                var Cmd = new SqlCommand(ConsultaLoc, Cnn);
                var Lector = Cmd.ExecuteReader();
                while (Lector.Read())
                {
                    var UnaLocalidad = new LocalidadEN();
                    UnaLocalidad.CodLoc = Conversions.ToInteger(Lector[0]);
                    UnaLocalidad.CodProvincia = Conversions.ToInteger(Lector[1]);
                    UnaLocalidad.Descripcion = Conversions.ToString(Lector[2]);
                    UnaLocalidad.CodigoPostal = Conversions.ToString(Lector[3]);
                    ListaLoc.Add(UnaLocalidad);
                }

                return ListaLoc;
            }
        }
    }
}